pyomo/core/kernel/conic.py

```diff
@@ -934,7 +934,7 @@ def as_domain(cls, r, x):
         b = block()
         b.r = variable_tuple([variable(lb=0) for i in range(len(r))])
         b.x = variable()
-        b.c = _build_linking_constraints(list(r) + [x], list(b.r) + [x])
+        b.c = _build_linking_constraints(list(r) + [x], list(b.r) + [b.x])
```

Comment:
This seems like a perfect case to add a small test to ensure correct behavior in different solvers.

Reply:
I added a test for the constraint structure. The behavior isn't really specific to mosek, it just happened to fail that way because mosek doesn't like constraints that specify the same variable twice.